Fill.py

```diff
@@ -499,16 +499,42 @@ def mark_for_locking(location: Location):
     single_player = multiworld.players == 1 and not multiworld.groups
 
     if prioritylocations:
+        regular_progression = []
+        deprioritized_progression = []
+        for item in progitempool:
+            if item.deprioritized:
+                deprioritized_progression.append(item)
+            else:
+                regular_progression.append(item)
+
         # "priority fill"
-        fill_restrictive(multiworld, multiworld.state, prioritylocations, progitempool,
+        # try without deprioritized items in the mix at all
+        fill_restrictive(multiworld, multiworld.state, prioritylocations, regular_progression,
                          single_player_placement=single_player, swap=False, on_place=mark_for_locking,
                          name="Priority", one_item_per_player=True, allow_partial=True)
 
-        if prioritylocations:
+        if prioritylocations and regular_progression:
             # retry with one_item_per_player off because some priority fills can fail to fill with that optimization
-            fill_restrictive(multiworld, multiworld.state, prioritylocations, progitempool,
-                            single_player_placement=single_player, swap=False, on_place=mark_for_locking,
-                            name="Priority Retry", one_item_per_player=False)
+            fill_restrictive(multiworld, multiworld.state, prioritylocations, regular_progression,
+                             single_player_placement=single_player, swap=False, on_place=mark_for_locking,
+                             name="Priority Retry", one_item_per_player=False, allow_partial=True)
+
+        if prioritylocations and deprioritized_progression:
+            # There are no more regular progression items that can be placed on any priority locations.
+            # We'd still prefer to place deprioritized progression items on priority locations over filler items.
+            fill_restrictive(multiworld, multiworld.state, prioritylocations, regular_progression,
+                             single_player_placement=single_player, swap=False, on_place=mark_for_locking,
+                             name="Priority Retry 2", one_item_per_player=False, allow_partial=True)
+
+        if prioritylocations and deprioritized_progression:
+            # retry with deprioritized items AND without one_item_per_player optimisation
+            fill_restrictive(multiworld, multiworld.state, prioritylocations, regular_progression,
+                             single_player_placement=single_player, swap=False, on_place=mark_for_locking,
+                             name="Priority Retry 3", one_item_per_player=False)
```

Comment:
This line not having `allow_partial=True` generates a bunch of locations that can't be filled at random. Not entirely sure what's going on, Fill,.py is way out of my league but at the very least I can comment from my experience modifying some lines a bit.

Reply:
Generates a bunch of locations? Uhhh

Reply:
I mean it might just not be solvable, at which point it's supposed to crash (prio fill not finding a valid layout is considered worth erroring on according to the current code)

Reply:
Makes sense. I ran a generation with games from my weekend session and had 181 priority locations, 607 regular progression items and 458 deprioritized progression. Without `allow_partial=True` it fails to place all 458 deprioritized progression. With it sometimes it fails, but not with the entire deprioritized pool and overall feels more consistent as it doesn't crash as often from what I've been testing.

Reply:
In fact, since this is currently just trying with `regular_progression` again (see: your other comment), this call to fill_restrictive is actually *guaranteed* to fail if it is reached, haha

Reply:
I ran another set of tests and noticed that the fill_restrictive calls after the first one never places items in priority locations. They return the same amount remaining priority locations betweel calls. Could this be a world issue? I'm getting it from Metroid Prime, ActRaiser and Minish Cap consistently. This consistently causes the crash I've been experimenting and removing those games from my Players folder properly fills everything.

Reply:
I'll mess about with it a bit. I don't really understand what the difference would be though